docs/zq/functions/cast.md

```diff
@@ -0,0 +1,60 @@
+### Function
+
+&emsp; **cast** &mdash; coerce a value to a different type
+
+### Synopsis
+
+```
+cast(val: any, t: type) -> any
+```
+
+### Description
+
+The _cast_ function performs type casts but handles both primitive types and
+complex types.  If the input type `t` is a primitive type, then the result
+is equivalent to
+```
+<t>(<val>)
+```
+e.g., the result of `cast(1, <string>)` is the same as `string(1)` which is `"1"`.
+
+For complex types, the cast function visits each leaf value in `val` and
+casts that value to the corresponding type in `t`.
+When a complex value has multiple levels of nesting,
+casting is applied recursively down the tree.  For example, cast is recursively
+applied to each element in array of records and recursively applied to each record.
```

Comment:
`cast(val, t)` doesn't walk `val` like this, though I think we do want to add a variant that does.

Reply:
I will clarify.  It's just referring to the idea that for complex types it casts each component of the complex type.  This came from Henri's description in the issue.